### DISCUSSION TITLE
Should a separate package be created for Hypixel Objects?

### DISCUSSION THREAD

**Discussion by unpluggedsam** [hypixelapiwrapper/hypixel-api-java#24] [Q&A]
As @TheNullicorn and I were meeting, we discussed whether classes (and the packages that contains them) used to obtain information from the API should be refactored into their own package. (eg: games, guild, ect.) 

The argument for moving the packages into a centralized package is so when an API user is trying to get a feel for the API and checks out the repository they can enter the package containing all of those classes, and know that that's where all the classes that they will need are located. They won't have the opportunity to get confused by looking through packages like http and log to see if there is anything hidden. 

The argument against moving the packages is:

  1. It's unnecessary. 
  2. It would lengthen import statements.
  3. It would create an abstraction that would not be beneficial.

We would love to hear the community's input and get more opinions.

**Comment by Mqlvin** [comment]
I think `guild` and `player` should have their own packages, containing all classes.

For `games`, I do still believe it'd be best to store all games in a separate package.
However, if you don't like that idea, we could move the classes from all 1-class packages to the main `games` package (sorry if that wasn't clear).

Contribution and user-ease should be a priority, so lengthening import statements isn't a major concern.

**Comment by Netwyn** [comment]
Yes, I think they should, but all game-related stuff should be stored in the `games` package.
Long import statements don't really matter, because most people use IDEs that auto-import classes anyway.

**Comment by TheNullicorn** [comment]
Just to elaborate on my thoughts a little more...

The library we're developing *is* a <ins>Hypixel</ins> API <ins>wrapper</ins>. Having a `wrapper` or `hypixel` package inside of that is redundant to me, and our temporary base package highlights that even more (double-hypixel: `io.github.hypixel_api_wrapper.hypixel` or double-wrapper: `io.github.hypixel_api_wrapper.wrapper`). If a package doesn't add anything to the organization that came before it, I don't believe it adds anything to the structure of the project itself. Right now it's just obscuring what'll be our most used classes one layer deeper into the hierarchy.

**Comment by unpluggedsam** [comment]
Well, all of the objects inside the `guild`, `player`, and `games` packages are somewhat of `api_retrieval_objects` so that's a possibility. I think we need to come up of a name that accurately and concisely summarizes the relationship of all the objects.
